File: backend/src/model/contest.rs
use diesel::prelude::*;
use serde::{Serialize, Deserialize};
use crate::schema::contest;

#[derive(Serialize, Deserialize, Clone, Debug, Queryable, AsChangeset, Identifiable)]
#[diesel(table_name = contest)]
pub struct Contest {
    pub id: i32,
    pub name: String,
    pub description: Option<String>,
    pub uid: i32
}

#[derive(Serialize, Deserialize, Clone, Debug, Queryable, AsChangeset, Identifiable)]
#[diesel(table_name = contest)]
pub struct UpdContest {
    pub id: i32,
    pub name: String,
    pub description: Option<String>
}

#[derive(Serialize, Deserialize, Clone, Debug, Insertable)]
#[diesel(table_name = contest)]
pub struct NewContest {
    pub name: String,
    pub description: Option<String>,
    pub uid: Option<i32>
}

File: problem-arena-frontend/src/model/contest.ts
export interface Contest {
    id: number
    name: string
    description: string
}

export interface NewContest {
    name: string
    description: string
}

export interface ContestDTO {
    id: number
    name: string
    description: string
    cnt: number
}

export interface ContestWithCreatorDTO {
    id: number
    name: string
    description: string
    cnt: number
    uid: number
    creator: string
}
